### PR TITLE
Versioning `diplomat-tool` through Cargo

### PR DESCRIPTION
**Pull request by robertbastian**
This is cleaner than relying on the globally installed `diplomat-tool` being the correct version, and allows for faster iteration.

Even after `cargo clean`, `cargo run -p diplomat-gen` is faster than `cargo install diplomat-tool && diplomat-tool`, and with incremental builds the rebuild is negligible but guarantees the correct version.

Uses https://github.com/rust-diplomat/diplomat/pull/348

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -228,12 +229,12 @@ icu_benchmark_macros = { path = "tools/benchmark/macros" }
 # ffi_coverage never used as a dep
 # testdata-scripts never used as a dep
 
-# Run `cargo make diplomat-install` to get the right diplomat binary installed
 # The version here can either be a `version = ".."` spec or `git = "https://github.com/rust-diplomat/diplomat", rev = ".."`
 # Diplomat must be published preceding a new ICU4X release but may use git versions in between
-diplomat = { version = "0.6" }
-diplomat-runtime = { version = "0.6" }
-diplomat_core = { version = "0.6" }
+diplomat = { git = "https://github.com/rust-diplomat/diplomat", branch = "lib" }
```
Comment (robertbastian) [inline Cargo.toml:234]: blocked on merging https://github.com/rust-diplomat/diplomat/pull/348

**Comment by Manishearth** [comment]
Not a huge fan since diplomat gets reinstalled once every few months, whereas I end up cleaning somewhat regularly because our feature combinations take up disk space, and `diplomat-gen` is something I need to run on most PRs.

**Comment by robertbastian** [comment]
> since diplomat gets reinstalled once every few months

I want to iterate faster than that.

> diplomat-gen is something I need to run on most PRs

It takes around 10 seconds for a clean build.

**Comment by Manishearth** [comment]
> It takes around 10 seconds for a clean build.

Ah. I feel like it takes longer to build but perhaps that's release mode. That's fine then

**Comment by robertbastian** [comment]
Yeah it takes minutes in release mode, which is why I don't want to keep reinstalling it that way.

File: tools/make/ffi.toml [hunk 2]
```diff
@@ -185,75 +150,6 @@ end
 
 # Install tasks
 
-# These tasks need a minimum version of cargo-make. Configs within cargo-make
-# task stanzas such as this allow setting the minimum version of deps (including
-# min version for cargo-make itself). But cargo-make might only be applying
-# such dep upgrades defined in the `install_crate` field after the task script
-# has run.
-[tasks.cargo-make-min-version]
```
Comment (Manishearth) [inline tools/make/ffi.toml:193]: nit: I'd like to keep these around, being able to diplomat-install is still somewhat convenient during debugging

**Review by Manishearth — APPROVED** [review]

File: tools/make/ffi.toml (hunk 2 above)
Comment (robertbastian) [reply tools/make/ffi.toml:193]: But we don't need `diplomat-tool` anymore

**Review by Manishearth — APPROVED** [review]

**Pull request merged**